### PR TITLE
fix #596 - outdated documentation for setup coq-prog-xxx.

### LINKED ISSUE #596: Broken (possibly outdated?) setup instructions regarding `coq-prog-name`
The instructions in https://proofgeneral.github.io/doc/master/userman/Obtaining-and-Installing/ instruct users to add `-emacs` to their `coq-prog-name` variable:
https://github.com/ProofGeneral/PG/blob/cc5bfd0b5524313313c8e394656cdb5321b62a42/doc/ProofGeneral.texi#L5976

It seems this actually breaks PG, leading to failures such as 
```
proof-activate-scripting: Wrong type argument: stringp, nil
```
or
```
proof-activate-scripting: Starting process: coqtop -emacs -emacs <..more arguments..>...failed
```
on newer versions of PG.

### PR DESCRIPTION
**Pull request by Matafou**
Fix #596

I did a refresh on this part of the documentation.
I propose to merge this before adapting maybe according to #590.

### CODE REVIEW

**Comment by Matafou** [comment]
I will merg e this soon if noone objects.

File: doc/ProofGeneral.texi [hunk 1]
```diff
@@ -4105,93 +4105,62 @@ the file.
...
 And then the right call to make will be done if you use the @kbd{M-x
-compile} command. Note that the lines are commented in order to be
-ignored by the proof assistant. It is possible to use this mechanism for
-all variables, @inforef{File Variables, ,emacs}.
+compile} command, and the correct @code{coqtop} will be called by
+ProofGeneral. Note that the lines are commented in order to be ignored
+by the proof assistant. It is possible to use this mechanism for all
+variables, @inforef{File Variables, ,emacs}.
+
+NOTE: @code{coq-prog-name} should contain only the @code{coqtop}
```
Comment (erikmd) [inline doc/ProofGeneral.texi:4145]: Nitpick/`NOTE:`: you may want to write `@emph{Note:}` like the other occurrence below?

File: doc/ProofGeneral.texi [hunk 2]
```diff
@@ -4217,16 +4186,6 @@ minor mode, type @kbd{M-x abbrev-mode RET}. When you are not in Abbrev
 mode you can expand an abbreviation by pressing @kbd{C-x '}
 (@code{expand-abbrev}). See the Emacs manual for more details.
 
-Coq Proof General has a special experimental feature called "Holes"
-which makes use of the abbreviation mechanism and includes a large list
-of command abbreviations. @xref{Holes feature}, for details.  With other
-provers, you may use the standard Emacs commands above to set up your
-own abbreviation tables.
```
Comment (erikmd) [inline doc/ProofGeneral.texi:4224]: Are you sure you want to remove this documentation from now on? 😉
but maybe you'd have plans to port part of the features of Holes to the same "elisp backend" as that of company-coq?

File: doc/ProofGeneral.texi [hunk 3]
```diff
@@ -4404,14 +4363,13 @@ Inserts ``End <section-name>.'' (this should work well with nested sections).
...
-file path'. The project file is typically named
+file path (-R,-Q,-I options). The project file is typically named
 @code{_CoqProject} and must be located at the directory root of
 your Coq project. Proof General searches for the Coq project file
 starting at the current directory and walking the directory
 structure upwards. The Coq project file contains the common
 options (especially @code{-R}) and a list of the files of the
-project, see the Coq reference manual, Section 15.3, ``Creating a
-Makefile for Coq modules''.
+project, see the Coq reference manual, Section @uref{https://coq.inria.fr/distrib/current/refman/practical-tools/utilities.html?highlight=makefile#building-a-coq-project,``Building a Coq project''}.
```
Comment (erikmd) [inline doc/ProofGeneral.texi:4372]: minor suggestion:

```suggestion
project, see the Coq reference manual, Section @uref{https://coq.inria.fr/distrib/current/refman/practical-tools/utilities.html#building-a-coq-project,``Building a Coq project''}.
```

File: doc/ProofGeneral.texi [hunk 4]
```diff
@@ -4439,6 +4398,12 @@ For backward compatibility, one can also configure the load path
 with the option @code{coq-load-path}, but this is not compatible
 with @code{CoqIde} or @code{coq_makefile}.
 
+NOTE: the Coq project file cannot define which version of @code{coqtop}
```
Comment (erikmd) [inline doc/ProofGeneral.texi:4401]: maybe:
```suggestion
@emph{Note:} the Coq project file cannot define which version of @code{coqtop}
```

**Review by erikmd — APPROVED** [review]
Thanks @Matafou; I did a quick review and left very minor suggestions

**Comment by erikmd** [comment]
So feel free @Matafou to merge this when you see fit.

BTW, small remark: it happens there are 3 PR merging strategies in GitHub (`Merge`, `Merge/Squash`, `Merge/Rebase`),
and I believe you have recently used the `Merge/Rebase` one? (maybe)

It seems actually that `Merge/Rebase` has two drawbacks:
* there is no mention anywhere of the PR number in the resulting commits messages;
* and the resulting commits are not GPG-signed (cf. the missing `Verified` slug on those commits).

However, `Merge` and `Merge/Squash` are very fine − `Merge/Squash` being especially fine if the PR contains only one commit I guess.

So, maybe for merging upcoming PRs we should only use `Merge` or `Merge/Squash` 😅

(I guess I could copy this suggestion in the wiki later on? Cc @cpitclaudel @hendriktews @Matafou @monnier FYI)

**Comment by erikmd** [comment]
> (I guess I could copy this suggestion in the wiki later on? Cc @cpitclaudel @hendriktews @Matafou @monnier FYI)

Done FYI: https://github.com/ProofGeneral/PG/wiki/Checklist-for-testing-and-reviewing-a-PR#merging-a-pr

File: doc/ProofGeneral.texi (hunk 2 above)
Comment (Matafou) [reply doc/ProofGeneral.texi:4224]: Indeed. I made a small poll on Zulip and it appear I was the only user of this feature.
I have tested yasnippet snippets for coq and it works quite well and it does not interfere with company coc completions (even if it also uses yasnippet). There are a few editing tricks that I will miss but nevermind.

File: doc/ProofGeneral.texi (hunk 1 above)
Comment (Matafou) [reply doc/ProofGeneral.texi:4145]: Will do

File: doc/ProofGeneral.texi (hunk 3 above)
Comment (Matafou) [reply doc/ProofGeneral.texi:4372]: Will do.

**Comment by Matafou** [comment]
Thanks @erikmd.

File: doc/ProofGeneral.texi (hunk 2 above)
Comment (Matafou) [reply doc/ProofGeneral.texi:4224]: More precisely I plan to

1. remove everything from the holes library shortly.
2. document and/or propose a set of user-customized abbreviations (unless this subsumed by company-coq maybe)
3. one day maybe try to reimplement the copy-paste-in-the-holes thing, even if nobody seems to be willing to try it...

**Comment by Matafou** [comment]
BTW @erikmd sorry for the rebase, this is the default "merge" button behaviour.

**Comment by erikmd** [comment]
No worries for this time ;)
Actually it's not the default action for me... so I believe the "default" action is just the last one used by the logged-in user. So to change it we need to use the drop-down list I guess.

**Pull request merged**